### ISSUE TITLE
Types: Types of referenced fields

### ISSUE THREAD

**Issue by mckelveygreg** [saiichihashimoto/sanity-typed#95]
Does the new @sanity-typed/types have a strategy to for referenced types?
For example, 
```ts
export const headlines = defineType({
  type: "document",
  name: "headline",
  title: "Headline",
  fields: [
    defineField({
      name: "categories",
      title: "Headline Category",
      type: "array",
      of: [
        defineArrayMember({
          type: "reference",
          to: [{ type: headlinesCategory.name }],
        }),
      ],
    }),
...
  ],
})

export type Headlines = Omit<
  Extract<SCHEMA_VALUES, { _type: "headline" }>,
  "categories"
> & {
  categories: Array<Extract<SCHEMA_VALUES, { _type: "headlinesCategory" }>>
}
```
Currently I am using the strategy from the previous iteration of the Schema builder, otherwise my types for `categories` are just the sanity reference values 🤷

**Comment by saiichihashimoto** [comment]
It doesn't right now. The reasoning is that the default `*[_type == 'documentType']` doesn't do anything unique with references. Technically, references [need the dereferencing operator](https://www.sanity.io/docs/how-queries-work#db43dfd18d7d) to expand. More importantly, how far should we go? Not all references should be expanded, especially when things get recursive. There's no good way to identify the proper type to be using and is entirely dependent on the groq that is used. We could bring in a helper that somehow allows you to specify what references to expand, but then we get close to mirroring the groq or the `Omit` stuff you have to do. I think this problem is better solved with a different solution: how should we handle groqs? Which opens a different can of worms. With this library's goal of being a drop-in replacement for sanity's `define*` methods in hopes that most of it can get merged in, all ideas here end up being way too much of a divergence.

TLDR: No, and there are no plans to. Any efforts towards this should instead be focused on a good groq solution that respects the schema and it's derived types.

Closing for now, unless you can think of something else.

**Comment by saiichihashimoto** [comment]
I AM starting to work on typing groq strings in an alpha branch and it'll have the dereferencing operator, so stay tuned...

https://github.com/saiichihashimoto/sanity-typed/pull/174

But also, now references technically carry the type it's referencing with them, so you could figure this out on your own:

https://github.com/saiichihashimoto/sanity-typed/pull/171
